arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/bookmark/BookmarkView.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.toolkit.bookmark
+
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.esri.arcgisruntime.mapping.Bookmark
+import com.esri.arcgisruntime.toolkit.R
+import kotlinx.android.synthetic.main.layout_bookmarkview.view.*
+
+class BookmarkView : FrameLayout {
+
+    var recyclerView: RecyclerView? = null
+
+    var bookmarksAdapter: BookmarkAdapter? = null
+
+    var onItemClickListener: OnItemClickListener<Bookmark>? = null
+
+    interface OnItemClickListener<T> {
```

Comment:
Is there any need for another interface?  if so, does it need to have a generic type `T` if we know the type of item is a `Bookmark`?

Reply:
we want the user to set the listener on the `BookmarkView` .  I have updated the type to `Bookmark`

Reply:
I was suggesting re-using `BookmarkAdapter.OnItemClickListener` instead but leave it here if you think it's cleaner.